src/protocols/WebSocket.ts

```diff
@@ -105,20 +105,24 @@ export default class WebSocketProtocol extends BaseProtocolRealtime {
       }
 
       this.client.onopen = () => {
-        this.clientConnected();
+        this.clientConnected().then(() => {
+          this.pingIntervalId = setInterval(() => {
+            this.ping();
+            this.pongTimeoutId = setTimeout(() => {
+              this.client.close();
+            }, this._pongTimeout);
+          }, this._pingInterval);
+        });
```

Comment:
Nitpicking (readability): I struggle a little bit to read this part

Reply:
read it again please I've changed the logic